File: frontend/src/services/ProjectService.js
import api from "./client/apiClient";

const createProject = (project) => {
  return api.post("/projects", project);
};

const updateProject = (id, project) => {
  return api.put(`/projects/${id}`, project);
};

const getProjectById = (id) => {
  return api.get(`/projects/${id}`);
};

const getAllProjects = () => {
  return api.get("/projects");
};

const deleteProject = (id) => {
  return api.delete(`/projects/${id}`);
};

const updateProjectStatus = (id, status) => {
  return api.put(`/projects/${id}/status`, { status });
};

export default {
  createProject,
  updateProject,
  getProjectById,
  getAllProjects,
  deleteProject,
  updateProjectStatus,
};

File: backend/src/main/java/org/dev/pys/service/ProjectService.java
package org.dev.pys.service;

import org.dev.pys.dto.request.CreateProjectRequest;
import org.dev.pys.dto.request.UpdateProjectRequest;
import org.dev.pys.dto.request.UpdateProjectStatusRequest;
import org.dev.pys.dto.response.ProjectResponse;

import java.util.List;

public interface ProjectService {

    ProjectResponse createProject(CreateProjectRequest request);

    ProjectResponse updateProject(Long id, UpdateProjectRequest request);

    ProjectResponse getProjectById(Long id);

    List<ProjectResponse> getAllProjects();

    void deleteProject(Long id);

    ProjectResponse updateProjectStatus(Long id, UpdateProjectStatusRequest request);
}
